gen_def.sh.in

```diff
@@ -0,0 +1,44 @@
+#! /bin/sh
+
+# overriding user localization settings for sort
+export LC_ALL=C
+
+HEADER='LIBRARY wpcap.dll\nEXPORTS'
+TMPFILE=DELETEME.c
+DEF=Win32/Prj/wpcap.def
+
+# The order matters. pcap/export-defs.h must come first
+cat pcap/export-defs.h pcap/pcap.h pcap/namedb.h nametoaddr.c > $TMPFILE
+
+# The order doesn't matter. Just don't include anything and make sure that the names
+# of all PCAP_API declarations come directly after the second space from the left.
+sed -i -e "s/#include <.*>//g" \
+       -e "s/#include ".*"//g" \
+       -e "s/const char/constchar/g" \
+       -e "s/const u_char/constu_char/g" \
+       -e "s/**pcap/pcap/g" \
+       -e "s/*pcap/pcap/g" \
+       -e "s/*bpf/bpf/g" \
+       -e "s/eproto_db\[\]/eproto_db/g" \
+       -e "s/struct pcap_samp/structpcap_samp/g" \
+       -e "s/struct pcap_stat/structpcap_statp/g" \
+       -e "s/struct	pcap_etherent/structpcap_etherent/g" \
+       -e "s/struct addrinfo/structaddrinfo/g" $TMPFILE
+
+echo $HEADER > $DEF
+
+# Run the preprocessor on $TMPFILE with *just* these things defined and do some
+# command-line kung-fu with the result.
+@CC@ -E -undef -D_WIN32 -DBUILDING_PCAP -DINET6 $TMPFILE \
```

Comment:
-undef, however, isn't in the SUS, and won't necessarily be supported by all compilers.

Where do you expect this to be used?  If the autotools version would be used only by people using GCC or a compatible compiler (MinGW, or a Linux/\*BSD/macOS box), you might want to have gen_def *not* be part of the "all" stage - the only people on a UN\*X box who would care would be 1) people preparing a source release and 2) people using the UN\*X box to generate the .def file.